shared/actions/favorite.js

```diff
@@ -2,18 +2,84 @@
 
 import engine from '../engine'
 import * as Constants from '../constants/favorite'
-
+import {badgeApp} from './notifications'
+import {canonicalizeUsernames, parseFolderNameToUsers} from '../util/kbfs'
+import _ from 'lodash'
 import type {Folder, favorite_favoriteList_rpc} from '../constants/types/flow-types'
 import type {Dispatch} from '../constants/types/flux'
-
 import type {FavoriteList} from '../constants/favorite'
+import type {Props as FolderProps} from '../folders/render'
+
+const folderToProps = (folders: Array<Folder>, username: string = ''): FolderProps => { // eslint-disable-line space-infix-ops
```

Comment:
moved this from the reducer into the action so we can also dispatch the badging call